project_euler/problem_064/sol1.py

```diff
@@ -0,0 +1,76 @@
+"""
+Project Euler Problem 64: https://projecteuler.net/problem=64
+
+All square roots are periodic when written as continued fractions.
+For example, let us consider sqrt(23).
+It can be seen that the sequence is repeating.
+For conciseness, we use the notation sqrt(23)=[4;(1,3,1,8)],
+to indicate that the block (1,3,1,8) repeats indefinitely.
+Exactly four continued fractions, for N<=13, have an odd period.
+How many continued fractions for N<=10000 have an odd period?
+
+References:
+- https://en.wikipedia.org/wiki/Continued_fraction
+"""
+
+from math import sqrt, floor
+
+
+def continuous_fraction_period(n: int) -> int:
+    """
+    Returns the continued fraction period of a number n.
+
+    >>> continuous_fraction_period(2)
+    1
+    >>> continuous_fraction_period(5)
+    1
+    >>> continuous_fraction_period(7)
+    4
+    >>> continuous_fraction_period(11)
+    2
+    >>> continuous_fraction_period(13)
+    5
+    """
+    numerator = 0.0
+    denominator = 1.0
+    ROOT = int(sqrt(n))
+    an = ROOT
+    period = 0
+    while an != 2*ROOT:
+        numerator = denominator*an - numerator
+        denominator = (n - numerator**2)/denominator
+        an = int((ROOT + numerator)/denominator)
+        period += 1
+    return period
+
+
+def solution(n: int = 10000) -> int:
+    """
+    Returns the count of numbers <= 10000 with odd periods.
+    This function calls continuous_fraction_period for numbers which are
+    not perfect squares.
+    This is checked in if sr - floor(sr) != 0 statement.
+    If an odd period is returned by continuous_fraction_period,
+    count_odd_periods is increased by 1.
+
+    >>> solution(2)
+    1
+    >>> solution(5)
+    2
+    >>> solution(7)
+    2
+    >>> solution(11)
+    3
+    >>> solution(13)
+    4
+    """
+    count_odd_periods = 0
+    for i in range(2, n+1):
+        sr = sqrt(i)
+        if sr - floor(sr) != 0:
+            if continuous_fraction_period(i) % 2 == 1:
+                count_odd_periods += 1
+    return count_odd_periods
+
+if __name__ == "__main__":
+    print(f"{solution(int(input().split()))}")
```

Comment:
```Suggestion
if __name__ == "__main__":
    print(solution())
```

Replace `an` with a descriptive variable name

Reply:
The program seems much more valuable if it is not hardcoded and continues to enable experimentation.  Why remove the end user's ability to experiment with different input values?

Reply:
On it with the changes!!

Reply:
> The program seems much more valuable if it is not hardcoded and continues to enable experimentation. Why remove the end user's ability to experiment with different input values?

I think this allows you to call the solution function both with and without a value. When called without value, it accepts the 10000 default value given in the function declaration.

Reply:
@kakashi215 Yes.  I did read the code.  I like `print(f"{solution(int(input().split()))}")` the way that it is now and do not see how `print(f"solution() = {solution()}")` is an improvement.

Reply:
@ruppysuppy i think the below code does the same and is preferred as the pythonic way.
print(f"{solution()=}")
So am changing it to that.
Lemme know if we need to change it.

Reply:
@cclauss 
This `print(f"{solution(int(input().split()))}")` would always need a value to be entered.
But this `print(f"solution() = {solution()}")` can work without any input as well. Although i agree that you'd have to enter custom value in code.

Reply:
@ruppysuppy Based on the comments am resolving the issue.